src/Maestro/tests/Maestro.ScenarioTests/EndToEndFlowLogic.cs

```diff
@@ -0,0 +1,352 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.Maestro.Client.Models;
+using Microsoft.Extensions.Primitives;
+using NUnit.Framework;
+
+namespace Maestro.ScenarioTests
+{
+    [TestFixture]
+    public class EndToEndFlowLogic : MaestroScenarioTestBase
+    {
+        private readonly string testRepo1Name = "maestro-test1";
+        private readonly string testRepo2Name = "maestro-test2";
+        private readonly string testRepo3Name = "maestro-test3";
+        private readonly string sourceBuildNumber = "654321";
+        private readonly string source2BuildNumber = "987654";
+        //private string sourceCommit = "123456";
+        private string sourceCommit = "cc1a27107a1f4c4bc5e2f796c5ef346f60abb404";
+        private string source2Commit = "789101";
+        private string sourceBranch = "master";
+        private readonly IImmutableList<AssetData> source1Assets;
+        private readonly IImmutableList<AssetData> source2Assets;
+        private readonly IImmutableList<AssetData> source1AssetsUpdated;
+        IImmutableList<AssetData> childSourceBuildAssets;
+        IImmutableList<AssetData> childSourceAssets;
+        private List<DependencyDetail> expectedDependenciesSource1;
+        private List<DependencyDetail> expectedDependenciesSource2;
+        private List<DependencyDetail> expectedDependenciesSource1Updated;
+        private List<DependencyDetail> expectedCoherencyDependencies;
+        private TestParameters _parameters;
```

Comment:
Wow... this is a LOT of fields.  I guess this is a side effect of these tests being so "big"?

Reply:
Yeah, it's because they're end to end tests - they cover a lot of ground. And because I'm sharing as much code as possible they all end up as fields instead of locals.